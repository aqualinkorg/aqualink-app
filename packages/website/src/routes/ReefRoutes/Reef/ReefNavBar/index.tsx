import React from "react";
import {
  AppBar,
  Toolbar,
  Grid,
  Typography,
  IconButton,
  Avatar,
  withStyles,
  WithStyles,
  createStyles,
  Theme,
} from "@material-ui/core";
import { Link } from "react-router-dom";
import ArrowBack from "@material-ui/icons/ArrowBack";

const ReefNavBar = ({
  reefName,
  lastSurvey,
  managerName,
  classes,
}: ReefNavBarProps) => (
  <AppBar className={classes.appBar} position="static">
    <Toolbar>
      <Grid item container justify="space-between" alignItems="center" xs={12}>
        <Grid item xs={5}>
          <Grid alignItems="center" direction="row" container spacing={1}>
            <Grid item>
              <Link style={{ color: "inherit" }} to="/reefs">
                <IconButton edge="start" color="inherit" aria-label="menu">
                  <ArrowBack />
                </IconButton>
              </Link>
            </Grid>
            {reefName && lastSurvey && (
              <Grid item xs={7} direction="column" container>
                <Grid item>
                  <Typography variant="h4">{reefName}</Typography>
                </Grid>
                <Grid item>
                  <Typography variant="subtitle1">{`Last surveyed: ${lastSurvey}`}</Typography>
                </Grid>
              </Grid>
            )}
          </Grid>
        </Grid>
        <Grid item xs={5}>
          {managerName && (
            <Grid container justify="flex-end" alignItems="center" spacing={2}>
              <Grid item>
                <Grid container direction="column" alignItems="flex-end">
                  <Typography variant="subtitle2">
                    {`Managed by: ${managerName}`}
                  </Typography>
                  <Typography variant="caption">Contact Manager</Typography>
                </Grid>
              </Grid>
              <Grid item>
                <Grid container>
                  <Avatar />
                </Grid>
              </Grid>
            </Grid>
          )}
        </Grid>
      </Grid>
    </Toolbar>
  </AppBar>
);

const styles = (theme: Theme) =>
  createStyles({
    appBar: {
      "&.MuiPaper-root": {
        backgroundColor: theme.palette.primary.main,
      },
    },
  });

interface ReefNavBarIncomingProps {
  reefName?: string;
  lastSurvey?: string;
  managerName?: string;
}

type ReefNavBarProps = ReefNavBarIncomingProps & WithStyles<typeof styles>;

export default withStyles(styles)(ReefNavBar);
