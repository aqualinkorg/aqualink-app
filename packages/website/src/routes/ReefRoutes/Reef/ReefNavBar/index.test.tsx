import React from "react";
import { render } from "@testing-library/react";
import ReefNavBar from ".";

jest.mock("react-router-dom", () => ({
  useHistory: () => ({
    push: jest.fn(),
  }),
}));

test("renders as expected", () => {
  const { container } = render(
    <ReefNavBar reefName="" managerName="" lastSurvey="May 10, 2020" />
  );
  expect(container).toMatchSnapshot();
});
