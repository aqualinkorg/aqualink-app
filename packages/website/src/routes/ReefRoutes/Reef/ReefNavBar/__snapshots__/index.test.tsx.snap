// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<div>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary ReefNavBar-appBar-1 MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-between MuiGrid-grid-xs-12"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-align-items-xs-center"
          >
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <a
                href="/reefs"
              >
                <button
                  aria-label="menu"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </a>
            </div>
            
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5"
        />
      </div>
    </div>
  </header>
</div>
`;
